Description: Use raise on click option.
Author: ?

Index: mutter-2.91.2/src/core/prefs.c
===================================================================
--- mutter-2.91.2.orig/src/core/prefs.c	2010-11-26 11:26:40.062551001 +1100
+++ mutter-2.91.2/src/core/prefs.c	2010-11-26 11:26:45.122551001 +1100
@@ -1515,10 +1515,7 @@
 gboolean
 meta_prefs_get_raise_on_click (void)
 {
-  /* Force raise_on_click on for click-to-focus, as requested by Havoc
-   * in #326156.
-   */
-  return raise_on_click || focus_mode == META_FOCUS_MODE_CLICK;
+  return raise_on_click;
 }
 
 const char*
