Index: mutter-2.91.2/src/core/window.c
===================================================================
--- mutter-2.91.2.orig/src/core/window.c	2010-11-26 11:33:29.302551001 +1100
+++ mutter-2.91.2/src/core/window.c	2010-11-26 11:33:30.962551001 +1100
@@ -1709,6 +1709,36 @@
 }
 
 gboolean
+meta_window_is_minimized (MetaWindow *window)
+{
+  return window->minimized;
+}
+
+gboolean
+meta_window_is_maximized (MetaWindow *window)
+{
+  return META_WINDOW_MAXIMIZED(window);
+}
+
+gboolean
+meta_window_is_maximized_horizontally (MetaWindow *window)
+{
+  return META_WINDOW_MAXIMIZED_HORIZONTALLY (window);
+}
+
+gboolean
+meta_window_is_maximized_vertically (MetaWindow *window)
+{
+  return META_WINDOW_MAXIMIZED_VERTICALLY (window);
+}
+
+gboolean
+meta_window_allows_resize (MetaWindow *window)
+{
+  return META_WINDOW_ALLOWS_RESIZE (window);
+}
+
+gboolean
 meta_window_showing_on_its_workspace (MetaWindow *window)
 {
   gboolean showing;
Index: mutter-2.91.2/src/include/window.h
===================================================================
--- mutter-2.91.2.orig/src/include/window.h	2010-08-24 06:21:06.000000000 +1000
+++ mutter-2.91.2/src/include/window.h	2010-11-26 11:33:30.962551001 +1100
@@ -89,6 +89,13 @@
 const char * meta_window_get_description (MetaWindow *window);
 const char * meta_window_get_wm_class (MetaWindow *window);
 const char * meta_window_get_wm_class_instance (MetaWindow *window);
+
+gboolean meta_window_is_minimized (MetaWindow *window);
+gboolean meta_window_is_maximized (MetaWindow *window);
+gboolean meta_window_is_maximized_horizontally (MetaWindow *window);
+gboolean meta_window_is_maximized_vertically (MetaWindow *window);
+gboolean meta_window_allows_resize (MetaWindow *window);
+
 /* Return whether the window would be showing if we were on its workspace */
 gboolean    meta_window_showing_on_its_workspace (MetaWindow *window);
 
