Index: mutter-2.91.4/src/core/display.c
===================================================================
--- mutter-2.91.4.orig/src/core/display.c	2010-12-23 10:09:54.436944000 +1100
+++ mutter-2.91.4/src/core/display.c	2010-12-23 10:11:05.876944001 +1100
@@ -131,6 +131,9 @@
 enum
 {
   OVERLAY_KEY,
+  OVERLAY_KEY_DOWN,
+  OVERLAY_KEY_WITH_MODIFIER,
+  OVERLAY_KEY_WITH_MODIFIER_DOWN,
   FOCUS_WINDOW,
   WINDOW_CREATED,
   WINDOW_DEMANDS_ATTENTION,
@@ -234,6 +237,33 @@
                   g_cclosure_marshal_VOID__VOID,
                   G_TYPE_NONE, 0);
 
+  display_signals[OVERLAY_KEY_DOWN] =
+    g_signal_new ("overlay-key-down",
+                  G_TYPE_FROM_CLASS (klass),
+                  G_SIGNAL_RUN_LAST,
+                  0,
+                  NULL, NULL,
+                  g_cclosure_marshal_VOID__VOID,
+                  G_TYPE_NONE, 0);
+
+  display_signals[OVERLAY_KEY_WITH_MODIFIER] =
+    g_signal_new ("overlay-key-with-modifier",
+                  G_TYPE_FROM_CLASS (klass),
+                  G_SIGNAL_RUN_LAST,
+                  0,
+                  NULL, NULL,
+                  g_cclosure_marshal_VOID__UINT,
+                  G_TYPE_NONE, 1, G_TYPE_UINT);
+
+  display_signals[OVERLAY_KEY_WITH_MODIFIER_DOWN] =
+    g_signal_new ("overlay-key-with-modifier-down",
+                  G_TYPE_FROM_CLASS (klass),
+                  G_SIGNAL_RUN_LAST,
+                  0,
+                  NULL, NULL,
+                  g_cclosure_marshal_VOID__UINT,
+                  G_TYPE_NONE, 1, G_TYPE_UINT);
+
   display_signals[WINDOW_CREATED] =
     g_signal_new ("window-created",
                   G_TYPE_FROM_CLASS (klass),
Index: mutter-2.91.4/src/core/keybindings.c
===================================================================
--- mutter-2.91.4.orig/src/core/keybindings.c	2010-12-23 10:09:56.826944001 +1100
+++ mutter-2.91.4/src/core/keybindings.c	2010-12-23 10:11:05.876944001 +1100
@@ -1241,13 +1241,33 @@
 {
   if (event->xkey.keycode != display->overlay_key_combo.keycode)
     {
+      if (display->overlay_key_only_pressed == TRUE)
+        {
+          // emit signal
+          if (event->xkey.type == KeyPress)
+            {
+              g_signal_emit_by_name (display, "overlay-key-with-modifier-down",
+                                     event->xkey.keycode);
       display->overlay_key_only_pressed = FALSE;
+            }
+        }
+      else
+        {
+          if (event->xkey.type == KeyRelease)
+            {
+              g_signal_emit_by_name (display, "overlay-key-with-modifier",
+                                     event->xkey.keycode);
+              display->overlay_key_only_pressed = TRUE;
+            }
+        }
+
       return FALSE;
     }
 
   if (event->xkey.type == KeyPress)
     {
       display->overlay_key_only_pressed = TRUE;
+      g_signal_emit_by_name (display, "overlay-key-down", NULL);
     }
   else if (event->xkey.type == KeyRelease && display->overlay_key_only_pressed)
     {
