Description: Fix unminimized windows not being mapped
Bug-Ubuntu: https://launchpad.net/bugs/620758

--- a/src/compositor/mutter-window.c	2010-07-02 13:14:39.000000000 -0400
+++ b/src/compositor/mutter-window.c	2010-08-26 03:03:09.034198001 -0400
 
@@ -1150,7 +1151,7 @@ mutter_window_show (MutterWindow   *self
       g_assert_not_reached();
     }
 
-  if (priv->redecorating ||
+  if (
       info->switch_workspace_in_progress ||
       event == 0 ||
       !start_simple_effect (self, event))
