Index: mutter-2.31.5/src/gdk-compat.h
===================================================================
--- mutter-2.31.5.orig/src/gdk-compat.h	2010-09-02 18:08:28.197537002 +0200
+++ mutter-2.31.5/src/gdk-compat.h	2010-09-02 18:08:45.065537001 +0200
@@ -9,10 +9,23 @@
  * using GDK_DISABLE_DEPRECATED.
  */
 
-#if !GTK_CHECK_VERSION (2, 21, 1)
-
+#define gdk_window_get_background(w,c)    *c = GDK_WINDOW_OBJECT (w)->bg_color
 #define gdk_visual_get_depth(v)           GDK_VISUAL(v)->depth
-
-#endif /*GTK_CHECK_VERSION */
+#define gdk_window_get_back_pixmap(w,p,r)                           \
+  G_STMT_START {                                                    \
+    GdkWindowObject *priv = GDK_WINDOW_OBJECT (w);                  \
+                                                                    \
+    if (p != NULL)                                                  \
+      {                                                             \
+        if (priv->bg_pixmap == GDK_PARENT_RELATIVE_BG ||            \
+            priv->bg_pixmap == GDK_NO_BG)                           \
+          *p = NULL;                                                \
+        else                                                        \
+          *p = priv->bg_pixmap;                                     \
+      }                                                             \
+                                                                    \
+    if (r != NULL)                                                  \
+      *r = (priv->bg_pixmap == GDK_PARENT_RELATIVE_BG);             \
+  } G_STMT_END
 
 #endif /* __GDK_COMPAT_H__ */
