Index: mutter-2.91.4/ltmain.sh
===================================================================
--- mutter-2.91.4.orig/ltmain.sh	2010-12-23 04:08:21.000000000 +1100
+++ mutter-2.91.4/ltmain.sh	2010-12-23 10:11:07.366944001 +1100
@@ -2566,6 +2566,11 @@
 	func_dirname "$file" "" "."
 	dir="$func_dirname_result"
 	;;
+        
+       -Wl,--as-needed)
+        deplibs="$deplibs $arg"
+        continue
+        ;;
 
       *)
 	func_warning "\`-dlopen' is ignored for non-libtool libraries and objects"
@@ -6147,6 +6152,15 @@
 	lib=
 	found=no
 	case $deplib in
+        -Wl,--as-needed)
+          if test "$linkmode,$pass" = "prog,link"; then
+            compile_deplibs="$deplib $compile_deplibs"
+            finalize_deplibs="$deplib $finalize_deplibs"
+          else
+            deplibs="$deplib $deplibs"
+          fi
+          continue
+          ;;
 	-mt|-mthreads|-kthread|-Kthread|-pthread|-pthreads|--thread-safe|-threads)
 	  if test "$linkmode,$pass" = "prog,link"; then
 	    compile_deplibs="$deplib $compile_deplibs"
